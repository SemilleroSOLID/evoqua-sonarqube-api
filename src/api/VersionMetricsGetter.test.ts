import { VersionMetricsGetterImpl } from ".";

const { SONAR_URL, SONAR_USERNAME, SONAR_PASSWORD } = process.env;
const api = new VersionMetricsGetterImpl(SONAR_URL!);

beforeAll(() => api.login(SONAR_USERNAME!, SONAR_PASSWORD!));

it('getVersions', async () => {
  const projectId = 'SCAPE-app';
  const versions = await api.getVersions(projectId);
  expect(versions).toMatchSnapshot();
});

it('getVersionMetrics', async () => {
  const projectId = 'SCAPE-app';
  const version = '2021-05-07T09:42:24-0500';
  const metrics = [
    { key: 'code_smells', name: 'Code Smells' },
    { key: 'sqale_index', name: 'Technical Debt' },
  ];
  const versionMetrics =
    await api.getVersionMetrics(projectId, version, metrics);
  expect(versionMetrics).toMatchSnapshot();
});
