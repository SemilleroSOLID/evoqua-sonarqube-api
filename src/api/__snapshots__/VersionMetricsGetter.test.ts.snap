// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getVersionMetrics 1`] = `
Object {
  "metrics": Array [
    "Code Smells",
    "Technical Debt",
  ],
  "values": Array [
    14,
    53,
  ],
  "version": "2021-05-07T09:42:24-0500",
}
`;

exports[`getVersions 1`] = `
Array [
  "2021-07-29T23:39:01-0500",
  "2021-05-09T00:30:02-0500",
  "2021-05-07T09:42:24-0500",
]
`;
